components/IndoorDirections/TypesOfDirections/DifferentFloorDirections.js

```diff
@@ -22,13 +25,15 @@ export function DifferentFloorDirections(props) {
         }
     }
 
+    var rooms = floorFrom == 9 ? Hall9Coordinates() : props.rooms;
     const linesToElevator = [];
     pathToElevator.forEach((element, index) => {
         if (index < pathToElevator.length - 1) {
             linesToElevator.push({x1: rooms[element].nearestPoint.x, y1: rooms[element].nearestPoint.y, x2: rooms[getNextRoomElevator(index)].nearestPoint.x, y2: rooms[getNextRoomElevator(index)].nearestPoint.y})
         }
     });
 
+    rooms = floorTo == 9 ? Hall9Coordinates() : props.rooms;
```

Comment:
you have one equal between rooms and floorTo and two equals between 9, I think you should also verify the type between rooms and the other variable (ex: floorTo), and also for the consistency

Reply:
Rooms is Hall9Coordinates if floorTo is 9, otherwise, it's props.rooms (which is HallXCoordinates)